snapcraft/commands/lifecycle.py

```diff
@@ -74,6 +74,18 @@ def fill_parser(self, parser: "argparse.ArgumentParser") -> None:
             default=os.getenv("SNAPCRAFT_BUILD_FOR"),
             help="Set target architecture to build for",
         )
+        parser.add_argument(
+            "--http-proxy",
+            type=str,
+            default=os.getenv("http_proxy"),
+            help="Set http proxy",
+        )
+        parser.add_argument(
+            "--https-proxy",
+            type=str,
+            default=os.getenv("https_proxy"),
+            help="Set https proxy",
+        )
```

Comment:
Where are these values used? Are we just allowing the user to define them? (If the actual implementation is just setting the environment variable value and allowing it to pass to the inner instance I think we should do that)

Reply:
this is for the legacy execution to work. I believe I will let this in, but we need a task to enable it for core22

Reply:
Is this what you mean by "enabling the proxy"?
https://github.com/snapcore/snapcraft/blob/260214788af30ea055b3170afd97b91e556341ae/snapcraft_legacy/internal/build_providers/_base_provider.py#L510